src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -0,0 +1,608 @@
+    LM_T(LmtMongo, ("Subscription attrs: %s", attrs.toString().c_str()));
+  }
+}
+
+
+/* ****************************************************************************
+*
+* setCondsAndInitialNotify -
+*
+*/
+static void setCondsAndInitialNotify
+(
+  const SubscriptionUpdate&        subUp,
+  const BSONObj&                   subOrig,
+  const std::string&               tenant,
+  const std::vector<std::string>&  servicePathV,
+  const std::string&               xauthToken,
+  const std::string&               fiwareCorrelator,
+  BSONObjBuilder*                  b,
+  bool*                            notificationDone
+)
+{
+  // notification can be changed by true by setCondsAndInitialNotify()
+  *notificationDone = false;
+
+  if (subUp.subjectProvided)
+  {
+    std::string status;
+    if (subUp.statusProvided)
+    {
+      status = subUp.status;
+    }
+    else
+    {
+      status = subOrig.hasField(CSUB_STATUS)? getStringFieldF(subOrig, CSUB_STATUS) : STATUS_ACTIVE;
+    }
+
+    std::string url;
+    if (subUp.notificationProvided)
+    {
+      url = subUp.notification.httpInfo.url;
+    }
+    else
+    {
+      url = getStringFieldF(subOrig, CSUB_REFERENCE);
+    }
+
+    RenderFormat attrsFormat;
+    if (subUp.attrsFormatProvided)
+    {
+      attrsFormat = subUp.attrsFormat;
+    }
+    else
+    {
+      attrsFormat = subOrig.hasField(CSUB_FORMAT)? stringToRenderFormat(getStringFieldF(subOrig, CSUB_FORMAT)) : NGSI_V2_NORMALIZED;
+    }
+
+    bool notificationDone;
+    setCondsAndInitialNotify(subUp, subUp.id, status, url, attrsFormat,
+                             tenant, servicePathV, xauthToken, fiwareCorrelator,
+                             b, &notificationDone);
+  }
+  else
+  {    
+    BSONArray conds = getArrayFieldF(subOrig, CSUB_CONDITIONS);
+    b->append(CSUB_CONDITIONS, conds);
+    LM_T(LmtMongo, ("Subscription conditions: %s", conds.toString().c_str()));
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setCount -
+*
+*/
+static void setCount(long long inc, const BSONObj& subOrig, BSONObjBuilder* b)
+{
+  if (subOrig.hasField(CSUB_COUNT))
+  {
+    long long count = getIntOrLongFieldAsLongF(subOrig, CSUB_COUNT);
+    setCount(count + inc, b);
+  }
+  else
+  {
+    // In this case we only add if inc is different from 0
+    if (inc > 0)
+    {
+      setCount(inc, b);
+    }
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setLastNotification -
+*
+*/
+static void setLastNotification(const BSONObj& subOrig, CachedSubscription* subCacheP, BSONObjBuilder* b)
+{
+  // FIXME P1: if CSUB_LASTNOTIFICATION is not in the original doc, it will also not be in the new doc.
+  //           Is this necessary?
+  //           The implementation would get a lot simpler if we ALWAYS add CSUB_LASTNOTIFICATION and CSUB_COUNT
+  //           to 'newSub'
+  if (subOrig.hasField(CSUB_LASTNOTIFICATION))
```

Comment:
Here you could do:

    if (!subOrig.hasField(CSUB_LASTNOTIFICATION))
    {
      return
    }

And you save the extra intentation for the rest